packages/tasit-contracts/src/scripts/populateDecentralandContracts.js

```diff
@@ -23,35 +23,170 @@ import fs from "fs";
+const { address: MARKETPLACE_ADDRESS } = Marketplace;
+const { address: GNOSIS_SAFE_ADDRESS } = GnosisSafeInfo;
+
+ConfigLoader.setConfig(config);
+
+const [minterWallet, sellerWallet] = accounts;
+const { address: sellerAddress } = sellerWallet;
+
+const manaContract = new Mana(MANA_ADDRESS);
+const landContract = new Land(LAND_PROXY_ADDRESS);
+const estateContract = new Estate(ESTATE_ADDRESS);
+const marketplaceContract = new Decentraland(MARKETPLACE_ADDRESS);
+const gnosisSafeContract = new GnosisSafe(GNOSIS_SAFE_ADDRESS);
+
+const provider = ProviderFactory.getProvider();
+
+const cancelSellOrder = async (nftAddress, id) => {
+  const action = marketplaceContract.cancelOrder(
+    nftAddress,
+    `${id}`,
+    gasParams
+  );
+  await action.waitForNonceToUpdate();
+};
+
+const cancelOrdersOfEstatesWithoutImage = async estatesIds => {
+  const getImageDataFromEstateId = async id => {
+    const image = await fetch(
+      `https://api.decentraland.org/v1/estates/${id}/map.png`
+    );
+    const data = (await image.buffer()).toString("base64");
+    return data;
+  };
+
+  // Note: Estate with 5 is one of the estates with blank image
+  const blankImageData = await getImageDataFromEstateId(5);
+
+  marketplaceContract.setWallet(sellerWallet);
+  for (let id of estatesIds) {
+    const imageData = await getImageDataFromEstateId(id);
+
+    if (imageData === blankImageData) {
+      console.log(
+        `Removing order of estate (id: ${id}) because it's with a blank image.`
+      );
+
+      await cancelSellOrder(ESTATE_ADDRESS, id);
+    }
+  }
+};
+
+const extractParcelsFromEstates = estates => {
+  let estatesParcels = [];
+  estates.forEach(estate => {
+    const { parcels } = estate;
+    estatesParcels = [...estatesParcels, ...parcels];
   });
+  return estatesParcels;
+};
 
-  landContract.setWallet(contractOwnerWallet);
-  const assignAction = landContract.assignMultipleParcels(
-    xArray,
-    yArray,
-    beneficiaryAddress,
+const updateParcelsData = async parcels => {
+  console.log("Updating parcels with metadata...");
+
+  landContract.setWallet(sellerWallet);
+  for (let parcel of parcels) {
+    let { x, y, metadata: parcelName } = parcel;
+    if (parcelName && parcelName !== "") {
+      const updateAction = landContract.updateLandData(x, y, parcelName);
+      await updateAction.waitForNonceToUpdate();
+    }
+  }
+};
+
+const getIdsFromParcels = async parcels => {
+  const parcelIds = [];
+  for (let parcel of parcels) {
+    const { x, y } = parcel;
+    const id = await landContract.encodeTokenId(`${x}`, `${y}`);
+    parcelIds.push(id);
+  }
+
+  return parcelIds;
+};
+
+// Tech-debt: Use `assignMultipleParcels` to save gas cost.
+// The amount of parcels per call should be short enough to avoid out-of-gas.
+const createParcels = async allParcels => {
+  console.log("Creating parcels...");
+
+  let parcelIds = [];
+  for (let parcel of allParcels) {
+    try {
+      const id = await createParcel(parcel);
```

Comment:
We could also move the metadata step here

Reply:
Done.

Reply:
Thanks! Resolving